python-package/lightgbm/sklearn.py

```diff
@@ -333,6 +334,12 @@ def fit(self, X, y,
         """
         evals_result = {}
         params = self.get_params()
+        # sklearn interface has another naming convention
+        params['seed'] = params.pop('seed', params.pop('random_state'))
+        params.pop('random_state', None)  # Avoid unknown parameter error from `train`
```

Comment:
yes, it's used for function argument and should be evaled first. but you can test it, it should be a defined and fixed behavior.